### ISSUE TITLE
Website updates: Tags

### ISSUE THREAD

**Issue by CathPag** [cncf/glossary#1006]
@cjyabraham, as discussed over Slack, ideally, we'd want three different tag styles:

- Fundamentals or advanced
- Tech/concept/property
- The rest

Each tag type tells us a little bit about that word. We could use the style we have now + a lighter version of the same pink + a ghost button style. Generally, it would go from bold to ghost but I feel that the latter category tag should be the full-colored one. Maybe you can ask your UX expert what the best approach is.

Also, it'd be great to have all tags at the top so people could filter by category (e.g., I want to see all architecture-related terms"). Here again, a UX perspective might be valuable. Should there be three lines?

- Fundamentals, advanced tags on top
- Tech/concept/property tags in the middle
- Other tags at the bottom?

I can ask for UX advice if no one is available on your end.

**Comment by cjyabraham** [comment]
hi @CathPag, We've put our heads together over here and come up with the following as a possible design for listing these various tags. Let us know your thoughts:
![Glossary Mockup copy](https://user-images.githubusercontent.com/463160/183450794-8fcfee1a-6338-4ae6-b126-9cba8bab57c4.jpg)

**Comment by cjyabraham** [comment]
Hi @CathPag, we considered what you are suggesting here but feel it can be misunderstood as "Architecture" being selected while the other two tags are not. Also, the "Fundamental" and "Technology" tags seem ambiguous without any context. We thought the "Level:" and "Category:" helped make what they were referring to more clear. We can look for an alternate solution if you think our solution takes up too much space.

Regarding the dropdowns, I will split that into [a new issue here](https://github.com/cncf/glossary/issues/1188) so we can tackle that separately. From your mockup where you have the dropdowns at the top of the article, I'm not sure quite how you want them to work. Can you explain? The [Kubernetes search UI](https://kubernetes.io/docs/reference/glossary/?fundamental=true) made more sense to me.

**Comment by CathPag** [comment]
True. If we color code the dropdown like the tags, it would make that clear. Thoughts?

**Comment by cjyabraham** [comment]
@CathPag are you asking whether styling the dropdowns to be the same style as the respective tags would help provide context to the tags themselves? I think that would help that particular problem, however, we're still confused why there are dropdowns on the page itself and what they do. Are those to filter the list of terms in the sidebar? Would it help to have a call to discuss?

**Comment by cjyabraham** [comment]
Makes sense. I've added these use-cases to [the issue](https://github.com/cncf/glossary/issues/1188) along with some other ideas.

**Comment by cjyabraham** [comment]
Here's another design we considered for fitting in the 3 taxonomies. We do prefer the first one I presented, since it gives the tags appropriate context, however, this one has the advantage that it would take up less space if that's a priority:
![2022-08-08-102557@2x](https://user-images.githubusercontent.com/463160/184424108-c520a02f-ed43-470d-8223-6e0cef2e27a1.jpg)

**Comment by CathPag** [comment]
I really like the Kubernetes example and the tag filter behavior discussed in #1188. How would that be compatible with this?

**Comment by cjyabraham** [comment]
My thought was that under the search input we have "Advanced Search..." which takes people to the [Kubernetes-like power-user mode](https://kubernetes.io/docs/reference/glossary/?fundamental=true). We can also have each tag link to that page with the appropriate tag selected in the filter.

**Comment by CathPag** [comment]
We just had our maintainer meeting and thought it might be best if we simplify it. Let's mimic what Kubernetes does (all tags selectable at the top, not hidden under advanced). We'll add fundamental and property to the tags from the third category and remove advanced, concept, and technology. What do you think?

**Comment by cjyabraham** [comment]
Ok. Are you referring to [this Kubernetes page](https://kubernetes.io/docs/reference/glossary/?fundamental=true)?

**Comment by CathPag** [comment]
Correct

**Comment by cjyabraham** [comment]
Ok. Can you share a design of how you want this to look and work? I'm having trouble envisioning how [the Kubernetes UI](https://kubernetes.io/docs/reference/glossary/?fundamental=true) will integrate with our current design without being overwhelming.

**Comment by CathPag** [comment]
Ok, let me do that tomorrow.

**Comment by CathPag** [comment]
How about something like this?
<img width="930" alt="Screen Shot 2022-08-18 at 9 34 25 AM" src="https://user-images.githubusercontent.com/74001907/185408141-d9be15d7-425a-46b0-8188-56b44c821333.png">

You can then select the tags, and it will show a list with terms below. 

When you click on a term, you still see the tags on top, and you filter by tags again.
<img width="928" alt="Screen Shot 2022-08-18 at 9 36 47 AM" src="https://user-images.githubusercontent.com/74001907/185408676-763a9a35-d1ae-4d5a-b917-f2fb7ed94df7.png">

What do you think?

**Comment by CathPag** [comment]
...I didn't do that in the mockup, but tags should probably be in alphabetical order.

**Comment by cjyabraham** [comment]
So when you click on a tag filter, does it generate a new list of terms below it or does it filter the list of terms on the left pane? And do you see these filters showing up so prominently on every page of the site? I saw this feature as more just for power users so not something you'd want to be front and center...

**Comment by CathPag** [comment]
The selection would mimic the k8s docs behavior: 

- "ghost" tags on top
- selected tabs full color
- List of terms meeting that criteria below (left nav bar doesn't change)

You click on the term you want to read, which automatically deselects all tags. The user can now start a new filter. To keep the selected list, users would have to right-click and open a new tab.

How does that sound?

**Comment by cjyabraham** [comment]
I see. I did not see this use-case to be important enough to warrant these filters present on every page of the site. Also, as you add more filters in the future, it will continue to take up more real estate. I saw these filters more as something you'd want on an "Advanced Search" page or something. 

If your team is united, though, on the interface you've described, then I can implement it that way.

**Comment by cjyabraham** [comment]
I had a few more thoughts on my alternative proposal for consideration. The proposed filter-by-tag screen would be similar to that on the Kubernetes site and would allow for permalinking with tags in the querystring, like https://kubernetes.io/docs/reference/glossary/?fundamental=true.

People would access that page either by clicking on a tag (which would then load the page with that tag set to filter) or by clicking "Advanced search..." below the search bar in the left column. This might alternatively be labeled "Filter by tag..."

This would give people multiple obvious ways to access the tool without needing it to take up space on every page.

<img width="848" alt="Screen Shot 2022-09-05 at 1 09 24 PM" src="https://user-images.githubusercontent.com/463160/188494671-6a7b75d9-ed87-4779-ba70-7edab27e171c.png">

**Comment by CathPag** [comment]
Hmm...I wouldn't hide it in advanced search. It's always visible in the Kubernetes glossary, and that's great. People don't have to figure out that they can filter by tags, they see it right on the screen. 

I really see it as two ways to navigate the page:
1. per term (left nav bar)
2. per category ( top tags)

**Comment by iamNoah1** [comment]
@CathPag @cjyabraham I have no strong opinion here. Having it as close to the Kubernetes page as possible makes most sense for me.

**Comment by jihoon-seo** [comment]
- K8s glossary page:
  - the page **lists the terms** with a short description of each term.
  - there is no dedicated pages for each term.
  - selecting and deselecting tags **updates the list of terms** in responsive way.
- CNCF Glossary website:
  - each page is for showing term definition or some information (e.g. home page, how to contribute, contributor ladder, style guide).

Considering above, @CathPag could you draw another design proposal figure for the case:
from this,
<img width="928" alt="Screen Shot 2022-08-18 at 9 36 47 AM" src="https://user-images.githubusercontent.com/74001907/185408676-763a9a35-d1ae-4d5a-b917-f2fb7ed94df7.png">
when a user clicks one of the tags above?

**Comment by cjyabraham** [comment]
@CathPag while we're waiting on this, would you like me to proceed with the second part of this issue, that is, listing all tags equally below the title of the term, instead of having "Technology", "Concept", and "Property" above it?

**Comment by CathPag** [comment]
So sorry, @cjyabraham! We had our meeting last week (or the previous one), but I totally blanked. We would like to proceed with the mockup above with one slight change. The deselected tags are gray and become pink if selected by staying above. So, no tags below the term. Instead, the two active tags are pink above, while the rest is gray.

<img width="803" alt="Screen Shot 2022-09-28 at 9 51 40 AM" src="https://user-images.githubusercontent.com/74001907/192796901-a790b22b-5293-413d-80af-a24773773b33.png">


When people select the tags initially, a list of all available terms with those tags will be listed below. Just like in the Kubernetes glossary. Do you know what I mean?

**Comment by cjyabraham** [comment]
@CathPag can you respond to @jihoon-seo comment above? It'd be good to see the mockup he's requesting as I'm still having trouble envisioning the workflow. As he has outlined, there are significant differences to the k8s glossary and our site so simply copying over the same UX for this search-by-tag would likely not work. 

Could you also differentiate between a tag that is selected due to the article below it being tagged with it, a tag that is selected for the purposes of listing all terms tagged with it, and a tag that is unselected.

**Comment by CathPag** [comment]
Ooff...I don't have any of the mockups anymore. That means I have to recreate them. Really busy this week. I'll need more time.

**Comment by CathPag** [comment]
Ok, here you go:

Homepage with deselected tags:
<img width="955" alt="Screen Shot 2022-10-10 at 7 46 00 AM" src="https://user-images.githubusercontent.com/74001907/194859156-5cfa68af-2105-49a3-94e5-4aba910f121d.png">


Selected tags and list of options:
<img width="938" alt="Screen Shot 2022-10-10 at 7 46 35 AM" src="https://user-images.githubusercontent.com/74001907/194859258-ff5c1ad5-ef54-4ffe-9540-47511e0e031e.png">

Individual term with tags highlighted that correspond to this term
<img width="963" alt="Screen Shot 2022-10-10 at 7 47 11 AM" src="https://user-images.githubusercontent.com/74001907/194859387-08fce7f1-0c74-47b2-aa66-24732cd60f79.png">

What do you think?

**Comment by cjyabraham** [comment]
I don't like how the filter buttons also do double-service as a way to indicate which tags a term has been tagged with. I don't think this would be clear to me if I was a user of the site. 

@jihoon-seo do these screenshots help you evaluate the proposal against [this other option here](https://github.com/cncf/glossary/issues/1006#issuecomment-1237329626) which maintains the current tagging UI on [term pages](https://glossary.cncf.io/continuous-delivery/) and offers a filtering-by-tag interface on [the regular terms-of-tag page](https://glossary.cncf.io/tags/methodology/)?

**Comment by CathPag** [comment]
I just asked Andrea, a UX expert, to chime in.

**Comment by andreuxxxx** [comment]
Hello, I think that what you really need is a version of the tags called "chips". It's part of Material Design and a simple solution when the objective is to identify information in a group and filter it at the same time, [[here's the reference.]([url]]( https://material.io/components/chips#filter-chips)

Please let me know and I can do a mockup to envision how these may look in context.

**Comment by cjyabraham** [comment]
Thanks. Chips seem cool. I think, however, that we first need to take a step back and decide on the appropriate user-journey to this filter-by-tag functionality: should it be considered a "power-user" feature and so placed on an "Advanced search..." page [as described here](https://github.com/cncf/glossary/issues/1006#issuecomment-1237329626). Or should the filters be available on all pages of the site [as shown here](https://github.com/cncf/glossary/issues/1006#issuecomment-1273196808)?

**Comment by andreuxxxx** [comment]
Hello Chris! The fewer clicks to access a tool, the best. These chips have a double purpose, if the user decides not to filter with them, at the very least they will have information about what category are they in.  If these categories are highly discoverable, the most probable thing is that the user will be motivated to use them as filters. They may not be that interested in expending time searching for power features, though... 
I think adding a checkmark to the chips will more strongly hint the filter function: 
<img width="804" alt="image" src="https://user-images.githubusercontent.com/3219456/194979356-8cc5c989-cf20-4b24-b420-ea1f2a69be19.png">

**Comment by cjyabraham** [comment]
Do you have an example showing chips used for a double-purpose like this: both to indicate a piece of content is tagged with a particular tag and to operate a filterable list?

**Comment by andreuxxxx** [comment]
Alternatively, many docs and references use the right side of the screen to make discoverability easier. We can have a proper filtering tool there.  or in any other appropriate location. I can definitely help you to find the most elegant and efficient solution for that, but I would need some more time. Let me know if that works for you all. 
![image](https://user-images.githubusercontent.com/3219456/194982270-d0840ea2-6908-4ed0-b3ec-fb3c5fd2fca2.png)

**Comment by andreuxxxx** [comment]
> Do you have an example showing chips used for a double-purpose like this: both to indicate a piece of content is tagged with a particular tag and to operate a filterable list?

As for an example, [this is a different UI pattern and these categories don't cross](https://www.google.com/finance/), but it's using chips as descriptive tags and on the second click as tabs that divide content associated with them.
![image](https://user-images.githubusercontent.com/3219456/194990602-025c46ab-e26d-485d-b77b-95e86dd6ec7b.png)

**Comment by jihoon-seo** [comment]
> @jihoon-seo do these screenshots help you evaluate the proposal against https://github.com/cncf/glossary/issues/1006#issuecomment-1237329626 which maintains the current tagging UI on [term pages](https://glossary.cncf.io/continuous-delivery/) and offers a filtering-by-tag interface on [the regular terms-of-tag page](https://glossary.cncf.io/tags/methodology/)?

IMO:
- @cjyabraham's suggestion:
  - This option unavoidably introduces one more click.
  - nevertheless, this option seems feasible and implementable, and makes sense in terms of UI/UX for me.
- @CathPag's suggestion:
  - I couldn't see other sites that have chips used for a double-purpose.
    In above @andreuxxxx introduced [Google Finance](https://www.google.com/finance/), 
    but it seems that these chips are not used for a double-purpose (e.g., I cannot multi-select those chips),
    but to provide links to each page ('Indexes', 'Most active', 'Gainers', 'Losers', ...).
    Please let me know if someone finds one, then I might be able to get some hints by seeing the working example.
  - Suppose that a user is seeing the 'Abstraction' page:
    <img width="963" alt="Screen Shot 2022-10-10 at 7 47 11 AM" src="https://user-images.githubusercontent.com/74001907/194859387-08fce7f1-0c74-47b2-aa66-24732cd60f79.png">
    I guess that starting to toggle any chip will display the filtered term list like this:
    <img width="938" alt="Screen Shot 2022-10-10 at 7 46 35 AM" src="https://user-images.githubusercontent.com/74001907/194859258-ff5c1ad5-ef54-4ffe-9540-47511e0e031e.png">
    If we proceed with this option, my opinions on this are:
    - I think that it might be not appropriate in terms of UI/UX if the filtered term list appearing 1) pushes the term title ('Abstraction') and description downwards, or 2) eliminates the term title and description's visibility.
      Displaying the filtered term list should be undo-able.
    - Instead, we might use in-page hover popup, which is similar to [macOS's Spotlight Search popup](https://www.lifewire.com/thmb/1tN_Z8YtVW-9iWdxxlGrCfcYo4o=/2560x1448/filters:no_upscale():max_bytes(150000):strip_icc()/usingmacspotlight6-5c61746e46e0fb0001f25082.jpg):
      ![image](https://user-images.githubusercontent.com/46767780/195218314-2180c28a-5dc8-4786-8b06-f864c26a5368.png)
      In this figure, the black box is the in-page hover popup which has appeared because the user started to toggle chips,
      and if user clicks "X" on the right-top corner then the popup disappears and user goes back to the term 'Abstraction' page with only the tags of the term 'Abtraction' (which is 'Architecture' and 'Fundamental' in this example) selected.

**Comment by andreuxxxx** [comment]
Hello! I can do a full UX assessment with Card sorting and checking the existing IA, that's the only way we can create the best tool. I am happy to help with it. I will have some time until next week, though. If you are OK waiting a little bit, I am happy to invest the time to really do something that works. This would include  some usability testing and we can validate our ideas before executing. Let me know if this is an option!

**Comment by jihoon-seo** [comment]
@andreuxxxx For me, it sounds great. Thanks for your work!

**Comment by andreuxxxx** [comment]
Hello everyone, unfortunately, I won't have time to look deeper into this in the immediate term. That being said, Catherine's suggestion works well from a UX perspective. While there might be more elegant solutions, it's intuitive and will work very well for the user. I recommend implementing her suggestions. 
I would love to put more time an effort into this, as it's really exciting for me to collaborate with CNCF but I know I will only have availability after November. I think it's is best to get something done now.

**Comment by GarethAcidWorks** [comment]
Hi all, as mentioned I bumped into Catherine in Detroit we discussed the above. So lending a hand to hopefully push this through. I have re-created the filters / tags in Figma, linked and pasted below.

**The additions / recommendations I have given are:**
• Small title to add some context to the filters / tags on the glossary home page
• Back button included when required
• Colours and sizing to be optimised

Let me know if i can do anything more to help here, thanks

https://www.figma.com/file/lz5s64a0DAO6s13fzd2ViQ/CNCF_Glossary?node-id=202%3A439
![1  Glossary + Filter](https://user-images.githubusercontent.com/91599545/200615493-4c4b786f-e884-4aa4-bdf8-e175d14c98b8.jpg)
![2  Abstraction + Filter](https://user-images.githubusercontent.com/91599545/200615546-6f6b27d2-46e4-4159-9432-abac982ee644.jpg)

**Comment by CathPag** [comment]
Awesome, thank you, @GarethAcidWorks! A proper mockup for the one below is missing. I guess @cjyabraham can defer it from the two above: All filters are shown + the selected ones active which are gray on your second mockup, right? 

<img width="749" alt="Screen Shot 2022-11-09 at 9 04 45 AM" src="https://user-images.githubusercontent.com/74001907/200850930-53712c18-efa8-4f84-b225-5bc77d5ca837.png">

**Comment by CathPag** [comment]
Ok, great! Just wanted to be sure Chris knows what to do for this middle section. Thanks for your help! Can't wait to see this implemented!

**Comment by cjyabraham** [comment]
Thanks @GarethAcidWorks. Could you show a mockup of the tags assigned for an individual article? Currently we have them here:
<img width="664" alt="Screen Shot 2022-11-22 at 11 17 03 AM" src="https://user-images.githubusercontent.com/463160/203365772-e6edb7a2-bc98-4ff3-95c0-c559e55d05cb.png">

**Comment by GarethAcidWorks** [comment]
So I've added in the Property tag and made some optimisations to the to the additional tagging up top.

The hierarchy of the page appears to flow and make sense. Let me know if you think this works
<img width="1436" alt="Screenshot 2022-11-23 at 11 28 07 PM" src="https://user-images.githubusercontent.com/91599545/203631180-9345b464-3b37-4c81-9b31-fb6b94bd7199.png">

**Comment by cjyabraham** [comment]
From your screenshot, it's not clear to me that the Abstraction term is tagged with Architecture and Fundamental. They just look like separate nav elements above the article for browsing.

**Comment by GarethAcidWorks** [comment]
So to break it down, if we're going to introduce this system that needs to be fixed in it's position up top. Meaning:

a) Everything else needs to be moved down
or b) We blend the grey tag and the pink nav tag on the same line (please excuse my bad terms)
<img width="285" alt="Screenshot 2022-11-24 at 10 58 56 AM" src="https://user-images.githubusercontent.com/91599545/203715228-fede4060-826d-4a48-8a76-fa476a4529ae.png">

However, by putting them on the same line, and separating them shows they serve 2 purposes more clearly i feel. If it's impossible for one term to have a max number of tags... (for example 3) then this suggests we wont run into the issue of hz space issues. Although mobile may require some stacking.

<img width="720" alt="Screenshot 2022-11-24 at 10 56 36 AM" src="https://user-images.githubusercontent.com/91599545/203715590-7c7874fd-7366-475d-93cc-e5ba696a84df.png">

**Comment by CathPag** [comment]
Hi there, so we don't need property, concept, and technology anymore. We'll only use the new "tags." In fact, property will be a tag now.

**Comment by GarethAcidWorks** [comment]
Clear, so these below seem to be the logical next step, here's my thoughts on each:

1) Pink option is familiar but they look like buttons which can be interacted with, if that is not the case... we should avoid that confusion

2) Blue option is a reference to the 'Glossary colour' and suggests that these selected tabs are different to pink buttons, therefore the user should not be confused if they try to click.

3) Grey option - This one is my personal preference as i feel they clearly show something has been selected but they are not clickable, more so they help keep the user's attention on the content below

<img width="1435" alt="Screenshot 2022-11-29 at 12 03 17 PM" src="https://user-images.githubusercontent.com/91599545/204474697-b7ea9afc-1ee5-4d99-b678-2e419fdacfae.png">
<img width="1432" alt="Screenshot 2022-11-29 at 12 03 27 PM" src="https://user-images.githubusercontent.com/91599545/204474728-553fa2e0-cec3-41d0-9f98-875233855970.png">
<img width="1436" alt="Screenshot 2022-11-29 at 12 05 42 PM" src="https://user-images.githubusercontent.com/91599545/204474734-c0d75066-ed9f-46b3-8c46-876273d60bda.png">

**Comment by CathPag** [comment]
Gray works for me

**Comment by cjyabraham** [comment]
These screenshots seem to have reverted to where we were before, when I left [this comment](https://github.com/cncf/glossary/issues/1006#issuecomment-1325567761):
"From your screenshot, it's not clear to me that the Abstraction term is tagged with Architecture and Fundamental. They just look like separate nav elements above the article for browsing."

But now you say they are "not clickable", so I guess they don't work for browsing either?

I'm confused. @CathPag if your goal here is to get an interface where people can filter by tags, similar to [the Kubernetes page](https://kubernetes.io/docs/reference/glossary/?community=true&extension=true&fundamental=true), then let's just create a page where they can do that. Integrating the filter interface with the actual tags assigned to a term and having them present on every page of the site doesn't seem to work.

**Comment by CathPag** [comment]
Actually, it would be good if they were clickable (or selectable) so that users could see all, let's say, "architecture" related terms. 

This has been going on for so long it's really difficult to keep track of what we agreed on. Can we please make a push and do something in the next few days?  We can always adjust later on if something needs to be tweaked. Our current approach seems not to work, let's try to be more agile, please. We've been discussing this for months now...

Summary:

1. Get rid of "property," "concept," "technology"  
2. Home view with selectable (clickable/selectable) "tags"
<img width="806" alt="Screen Shot 2022-11-30 at 9 15 40 AM" src="https://user-images.githubusercontent.com/74001907/204819730-3da2d5ab-d621-4334-9a1d-9ee2b74fad0a.png">

Behavior: as people select a tab, terms start appearing below (same behavior as [Kubernetes Glossary](https://kubernetes.io/docs/reference/glossary/?core-object=true))

3. View of all terms with those tags (with the right looking tags, of course)
<img width="743" alt="Screen Shot 2022-11-30 at 9 15 55 AM" src="https://user-images.githubusercontent.com/74001907/204820204-b6c6791f-5f07-4afb-9984-db5aabb21993.png">

Note: the term itself should also be clickable (didn't make it a hyperlink in the mockup)

4. View of selected term with "tags" on top
<img width="794" alt="Screen Shot 2022-11-30 at 9 16 09 AM" src="https://user-images.githubusercontent.com/74001907/204820565-034a33fd-6a05-4b0e-b0c5-38b5fce152f7.png">

From here, the user can go back and view all terms with those two tags OR click on one tag only and view a selection of terms with that tag only. 

What do you think? Should we just do that and see how it works?

**Comment by cjyabraham** [comment]
I have removed the "property," "concept," "technology" category from the single term pages. I have not touched the data in the front matter for each term in case you want to migrate them into a normal tag or just delete.

**Comment by cjyabraham** [comment]
[This PR](https://github.com/cncf/glossary/pull/1715) has achieved many of the ideas listed in this issue. I'll close it for now and if there are further ideas for a subsequent iteration on this then please open a new clean issue.
